gen.go

```diff
@@ -80,14 +80,15 @@ func genPkg(odir string, p *bind.Package, lang string) error {
 	case "python3", "py3":
 		return fmt.Errorf("gopy: python-3 support not yet implemented")
 
-	case "go":
+	case "go", "gocffi":
 		o, err = os.Create(filepath.Join(odir, p.Name()+".go"))
 		if err != nil {
 			return err
 		}
 		defer o.Close()
 
-		err = bind.GenGo(o, fset, p, pyvers)
+		buildWithLibPython := lang != "gocffi"
+		err = bind.GenGo(o, fset, p, pyvers, buildWithLibPython)
```

Comment:
`pkg-config` influences the whole build process, I'd say that putting it under linking would be still opaque.

I realized that it is also not necessary to run `go tool cgo` to obtain the C header, as it is not needed to compile the CFFI binding.

See the updated commit which I like _somewhat_ better.

Reply:
so what about `capi` being either `"cpython"` or `"cffi"` ?
_ie:_
```go
func GenGo(w io.Writer, fset *token.FileSet, pkg *Package, lang int, capi string) error { ... }
```
(or perhaps, instead of a `string` a dedicated enum-like type?)

Reply:
Sounds reasonable, I'll update this later.

Reply:
I changed the `useLibPython` argument to `capi`, which can be either `"cffi"` or `"cpython"`.